tachyon/crypto/hashes/sponge/poseidon/poseidon_sponge_base.h

```diff
@@ -121,7 +121,7 @@ struct PoseidonSpongeBase : public FieldBasedCryptographicSponge<Derived> {
                                             size_t num_elements) const {
     const Derived& derived = static_cast<const Derived&>(*this);
 
-    std::vector<F> ret(num_elements);
+    std::vector<F> ret(num_elements, F::Zero());
```

Comment:
It seems like we don't need to initialize the values, since `SqueezeInternal` will be filled with the values. But just add a comment that why this is safe.